Content/Items/Misc/Accessories.BrokenGlasses.cs

```diff
@@ -0,0 +1,152 @@
+﻿using Mono.Cecil.Cil;
+using MonoMod.Cil;
+using StarlightRiver.Content.Items.BaseTypes;
+using StarlightRiver.Core.Systems.InstancedBuffSystem;
+using StarlightRiver.Helpers;
+using System;
+using Terraria.ID;
+
+namespace StarlightRiver.Content.Items.Misc
+{
+	public class BrokenGlasses : SmartAccessory
+	{
+		public override string Texture => AssetDirectory.MiscItem + Name;
+		public BrokenGlasses() : base("Broken Glasses", "Damage over time effects are able to critically strike") { }
+
+		public override void Load()
+		{
+			StarlightPlayer.OnHitNPCEvent += StarlightPlayer_OnHitNPCEvent;
+			StarlightPlayer.OnHitNPCWithProjEvent += StarlightPlayer_OnHitNPCWithProjEvent;
+		}
+
+		private void StarlightPlayer_OnHitNPCWithProjEvent(Player player, Projectile proj, NPC target, int damage, float knockback, bool crit)
+		{
+			if (Equipped(player))
+				BuffInflictor.Inflict(target, 600, new BrokenGlassesBuff() { lastInflicted = player });
+		}
+
+		private void StarlightPlayer_OnHitNPCEvent(Player player, Item Item, NPC target, int damage, float knockback, bool crit)
+		{
+			if (Equipped(player))
+				BuffInflictor.Inflict(target, 600, new BrokenGlassesBuff() { lastInflicted = player});
+		}
+
+		public override void SafeSetDefaults()
+		{
+			Item.rare = ItemRarityID.White; //funny reference :)	
+		}
+
+		public override void AddRecipes()
+		{
+			Recipe recipe = CreateRecipe();
+			recipe.AddIngredient(ItemID.Glass, 25);
+			recipe.AddRecipeGroup(RecipeGroupID.IronBar, 10);
+			recipe.AddTile(TileID.Anvils);
+			recipe.Register();
+		}
+	}
+
+	class BrokenGlassesBuff : InstancedBuff
+	{
+		public Player lastInflicted;
+
+		public override string Name => "BrokenGlassesBuff";
+
+		public override string DisplayName => "Broken Glasses";
+
+		public override string Texture => AssetDirectory.MiscItem + Name;
+
+		public override bool Debuff => true;
+
+		public override string Tooltip => "You cant see, loser!";
+	}
+
+	class BrokenGlassesSystem : IOrderedLoadable
```

Comment:
This should probably be rolled into the item class since it does nothing but patch one method